api_v2/src/leaves/leaves.service.ts

```diff
@@ -53,19 +54,27 @@ export class LeavesService {
     return toLeavesDTO(leaves, true);
   }
 
-  async createLeave(input: CreateLeaveRequestInput) {
+  async createLeave(input: CreateLeaveRequestInput): Promise<Leave[]> {
     try {
       await this.validateLeaveInput(input);
-      const newLeaves = await this.leavesRepository.createLeave(input);
 
-      // Create associated multi-projects for each leave
+      const newLeaves = await this.leavesRepository.createLeave(input);
+      // Create associated multi-projects
       for (const leave of newLeaves) {
```

Comment:
Maybe we can use Promise.all here for parallel execution
```suggestion
      await Promise.all(
        newLeaves.map((leave) =>
        this.multiProjectRepository.createForLeave(
          leave.id,
          input.leaveProjects
        )
      )
    );
```